lua/entities/gmod_wire_expression2/core/egpfunctions.lua

```diff
@@ -1035,15 +1038,25 @@ e2function number wirelink:egpHasObject( index )
 	return bool and 1 or 0
 end
 
+__e2setcost(20)
+
 --- Returns 1 if the object with specified index contains the specified point.
 e2function number wirelink:egpObjectContainsPoint(number index, vector2 point)
 	local bool, _, object = EGP:HasObject(this, index)
-	local _, pos = EGP:GetGlobalPos(this, index)
-	if this.TopLeft and object.CanTopLeft and object.w and object.h then
-		pos.x = pos.x + object.w / 2
-		pos.y = pos.y + object.h / 2
+	if bool then 
+		local hasVertices, pos = EGP:GetGlobalPos(this, index)
+		if this.TopLeft and object.CanTopLeft and object.w and object.h then
+			pos.x = pos.x + object.w / 2
+			pos.y = pos.y + object.h / 2
+		elseif hasVertices then
+			-- Localize the point to the object. Yeah.
+			local x, y = getCenterFromPos(pos)
+			local fakex, fakey = getCenter(makeArray(object), true)
+			return object:Contains({ x = point[1] - x + fakex, y = point[2] - y + fakey, angle = 0 }) and 1 or 0
+		end
+		return object:Contains({ x = point[1] - pos.x, y = point[2] - pos.y, angle = pos.angle }) and 1 or 0
```

Comment:
Couldn't all of this just be replaced with `object:Contains` and the individual logic be performed in the object Contains functions?

Reply:
Will look into it.

Reply:
I wanted to preserve the signature of `Contains` only taking a single value instead of having to pass `this` along with it. For an unparented polygon, the centroid is negated; it's only used for localizing the position of the point to the poly. Unless you're talking about the `poly.Contains` function itself.

Reply:
What I mean is, there shouldn't be an if-else tree for each type of object. They should all implement a 'ContainsPoint' or whatever function that works for each type of object, and this function should just call ContainsPoint. An if-else tree is jank.